src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs

```diff
@@ -324,17 +324,20 @@ public async Task DuplicateAndClose_TcpServerHandler(AddressFamily addressFamily
                 if (sameProcess)
                 {
                     Task handlerCode = Task.Run(() => HandlerServerCode(_ipcPipeName));
-                    RunCommonHostLogic(Environment.ProcessId);
+                    await RunCommonHostLogic(Environment.ProcessId);
                     await handlerCode;
                 }
                 else
                 {
-                    using RemoteInvokeHandle hServerProc = RemoteExecutor.Invoke(HandlerServerCode, _ipcPipeName);
-                    RunCommonHostLogic(hServerProc.Process.Id);
+                    RemoteInvokeHandle hServerProc = RemoteExecutor.Invoke(HandlerServerCode, _ipcPipeName);
+                    await RunCommonHostLogic(hServerProc.Process.Id);
+                    await hServerProc.DisposeAsync();
                 }
 
-                void RunCommonHostLogic(int processId)
+                async Task RunCommonHostLogic(int processId)
                 {
+                    await Task.Yield();
```

Comment:
If the intention here was to ditch the sync context for the `WaitForConnection` call, you'd have to `ConfigureAwait(false)` or use something like
```c#
await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
```